plugin/pkg/shoot/dns/admission.go

```diff
@@ -163,29 +188,10 @@ func (d *DNS) Admit(ctx context.Context, a admission.Attributes, o admission.Obj
 		return nil
```

Comment:
I think we should not return here, in order to also promote the `unmanaged` provider as a `primary` one.